crates/ide/src/display/navigation_target.rs

```diff
@@ -44,7 +44,7 @@ pub struct NavigationTarget {
     /// Clients should place the cursor on this range when navigating to this target.
     pub focus_range: Option<TextRange>,
     pub name: SmolStr,
-    pub kind: Option<SymbolKind>,
+    pub kind: Option<StructureNodeKind>,
```

Comment:
Why do we need this change? This I think can remain a SymbolKind?

Reply:
Should I map `SymbolKind` to `StructureNodeKind::SymbolKind` or just make `to_proto::symbol_kind` `pub(crate)` again?

Reply:
I think the easiest fix to fit StructureNodeKind into existing code would be this:


```diff
diff --git a/crates/rust-analyzer/src/handlers.rs b/crates/rust-analyzer/src/handlers.rs
index 706a39dab..900221b99 100644
--- a/crates/rust-analyzer/src/handlers.rs
+++ b/crates/rust-analyzer/src/handlers.rs
@@ -289,7 +289,10 @@ pub(crate) fn handle_document_symbol(
         let doc_symbol = lsp_types::DocumentSymbol {
             name: symbol.label,
             detail: symbol.detail,
-            kind: to_proto::symbol_kind(symbol.kind),
+            kind:  match symbol.kind {
+                StructureNodeKind::SymbolKind(kind) => to_proto::symbol_kind(kind),
+                StructureNodeKind::Region => lsp_types::SymbolKind::Module
+            },
             tags: Some(tags),
             deprecated: Some(symbol.deprecated),
             range: to_proto::range(&line_index, symbol.node_range),
```

Reply:
That's kinda what `to_proto::structure_node_kind` does

Reply:
But this keeps the change local to only doment_symbol, and doesn't require change NavigationTarget and such. Perhaps I am misunderstanding something, but I think the overall diff here can be much smaller and avoid touching anything outside file_structure/document_symbol?

Reply:
Yep. Removed StructureNodeKind from NavigationTarget, and we are down to `file_structure` changes. Code above isn't used in `handlers.rs`, as it's implemented in `to_proto::structure_node_kind` directly